components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java

```diff
@@ -9747,8 +9747,8 @@ public static Map<String, EndpointSecurity> setEndpointSecurityForAPIProduct(API
             }
             return endpointSecurityMap;
         } catch (ParseException e) {
-            String errorMessage = "Error while parsing Endpoint Config json";
-            throw new APIManagementException(errorMessage, e, ExceptionCodes.ERROR_PARSING_ENDPOINT_CONFIG);
+            String errorMessage = "Error while parsing Endpoint Config json. Value is missing or invalid.";
```

Comment:
Since we already check whether the endpointConfig exists, this errorMessage is not entirely correct. Let's revert the log message and the error code change.

Reply:
Reverted with [cb4e85a](https://github.com/wso2/carbon-apimgt/pull/13471/commits/cb4e85ab6a0b5d432e10d1fb16e1a258d1ccc866)